quicklens/src/test/scala-3/com/softwaremill/quicklens/test/CompileTimeTest.scala

```diff
@@ -0,0 +1,27 @@
+package com.softwaremill.quicklens.test
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class CompileTimeTest extends AnyFlatSpec with Matchers {
+  // #114
+  it should "not compile for too long in case of chained modify invocations" in {
+    val start = System.currentTimeMillis()
+    assertDoesNotCompile("""
```

Comment:
While the code currently does not compile, would it really be a bug if it did? It does compile in Scala 2. I think `assertDoesNotCompile` should not be used here.

Reply:
it's in Scala3 sources as `assertDoesNotCompile` is a Scala3-only thing as far as I know (in ScalaTest).

As for this not compiling... I though that was the point showing the compiler timing issue (which is fixed here), but now I see there are two problems

Reply:
I've extracted this as a separate issue, https://github.com/softwaremill/quicklens/issues/116

Reply:
And changed the type here to an unrelated one, which definitely shouldn't work :)